src/pyhf/readxml.py

```diff
@@ -1,18 +1,28 @@
-from pyhf import schema
-from pyhf import compat
-from pyhf import exceptions
+from __future__ import annotations
 
 import logging
-
-from pathlib import Path
+import os
+import typing as T
```

Comment:
Any objection to just following what seems to be more standard and just importing everything we'll need from `typing`?

https://github.com/scikit-hep/hist/blob/208244597b1313a5623b5d41331d7056860bc830/src/hist/typing.py#L4

```suggestion
from typing import TYPE_CHECKING, Callable, Iterable, Tuple, Union
```